### ISSUE TITLE
How to train a multi-task model based on your code?

### ISSUE THREAD

**Issue by o0t1ng0o** [CVIU-CSU/M2MRF-Lesion-Segmentation#3]
Hi, @haotianliu001 
May I ask how to train a multi-task model based on your code?

**Comment by haotianll** [comment]
Hi, does "multi-task" mean multiple segmentation tasks? If so, I think it's fessible, but you need to modify the code.

This code follows the stardard structure of MMSegmentation. I think you need to determain a suitable label encoding format and write code to load it. It's also necessary to rewrite the loss computation procedure based on [this](https://github.com/CVIU-CSU/M2MRF-Lesion-Segmentation/blob/master/mmseg/models/decode_heads/decode_head.py#L169-L187), the testing pipeline [here](https://github.com/CVIU-CSU/M2MRF-Lesion-Segmentation/blob/master/mmseg/models/segmentors/encoder_decoder.py#L309-L329), and the evaluation function [here](https://github.com/CVIU-CSU/M2MRF-Lesion-Segmentation/blob/master/mmseg/datasets/my_custom.py#L22-L33).

**Comment by o0t1ng0o** [comment]
Thank you for your prompt reply.

My multi-task model may include a segmentation task and a classification task.
So I may need to modify the dataset part and add new modules for this. 
It seems that it is a little bit complicated to modify on your code that is based on mmcv.
Could you recommend a multi-lesion segmentation repository that does not based on mmcv?

**Comment by haotianll** [comment]
It's true. The mmcv framework is a bit complicated to use. 

Coincidencely, for the multi-lesion segmentation task, I also have a simpler implementation [here](https://github.com/haotianliu001/HRNet-Lesion), which is based on [HRNet-Semantic-Segmentation](https://github.com/HRNet/HRNet-Semantic-Segmentation). This repository is unofficial, but you may try it. And I think it's not hard to add a new classification branch in it.

Besides, you can also find many popular segmentation networks in [segmentation_models.pytorch](https://github.com/qubvel/segmentation_models.pytorch).

**Comment by o0t1ng0o** [comment]
Thank you very much!!!
Your repository is very useful!
May I ask can it reach the on-par performance of HRNet that is reported in your paper?

**Comment by haotianll** [comment]
You are welcome!
Yes. They can almost achieve the equal performance.

**Comment by o0t1ng0o** [comment]
That is goooooood! 😄

**Comment by haotianll** [comment]
👌
